dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -44,6 +44,7 @@ private ProcessingManager(Configuration config) throws InvalidConfigurationExcep
 
         programProcessors.addAll(Arrays.asList(
                 atomicBlocksAsLocks ? AtomicAsLock.fromConfig(config) : null,
+                p->p.getMemory().fixAddressValues(),
```

Comment:
I agree: if this is important, it should get a name. You could create a class that has only the  `fromConfig` (or just `create`) method and that returns the lambda. So basically its a class with a single static method that returns an anonymous `ProgramProcessor`.

Reply:
It already has the name `fixAddressValues`.  This way it has minimal overhead.  Wrapping it will be necessary, when `ProgramProcessor` gets more abstract methods, or when we define `@Options` for it.  Though, `ProcessingManager$lambda` is unreadable in the debugger.

Reply:
The implicit naming makes this necessary pass harder to identify IMO.
I would rather have this pass as part of the `Memory` class directly if you want the least coding overhead.
Something like
```
public static ProgramProcessor fixateMemoryLayoutPass() {
   return p -> p.getMemory().fixateAddressValues();
}
```
This way, the pass is less hidden because it is not only defined in the `ProcessingManager`, it makes it more clear that fixating the memory layout is actually a pass and you make it more obvious that `fixateAddressValues` is a method that should only be called once. Indeed, you can make `fixateAddressValues` private and only make it accessible via the static method.
At the end of the day, these are trivial things that can be changed when needed though.